### PR TITLE
Add suport for building emlinux-k510 with arm

### PR DESCRIPTION
**Pull request by Ponsuke07**
# Purpose of pull request

This PR supports for building emlinux-k510 with arm.

# Test
## How to test

1. build with beaglebone (or qemuarm)

```conf/auto.conf
MACHINE = "beaglebone"
DISTRO = "emlinux-k510"
```

```
$: bitbake core-image-weston
```

## Test result

Without this PR, you will get following error:

```
ERROR: linux-libc-headers-k510-gitAUTOINC+3ddbe9bf6a-r0 do_install: oe_multilib_header: Unable to find header asm/kvm.h.
```

### CODE REVIEW

File: recipes-kernel/linux-libc-headers/linux-libc-headers-k510.bb [hunk 1]
```diff
@@ -3,3 +3,7 @@ require recipes-kernel/linux-libc-headers/linux-libc-headers-base_git.bb
 LIC_FILES_CHKSUM = "file://COPYING;md5=6bc538ed5bd9a7fc9398086aedcd7e46"
 
 DEPENDS += " rsync-native"
+
+# "asm/kvm.h" has been removed by kernel commit:
+#   541ad0150ca4 [arm: Remove 32bit KVM host support]
+do_install_armmultilib_remove_arm = "asm/kvm.h"
```
Comment (hiraku-wfs) [inline recipes-kernel/linux-libc-headers/linux-libc-headers-k510.bb]: Does this work correctly?
The do_install_armmultilib is not a variable but a function.

Comment (Ponsuke07) [reply recipes-kernel/linux-libc-headers/linux-libc-headers-k510.bb]: Yes,  It work.
I have checked it like below.

beaglebone and qemuarm
```
$: bitbake -e linux-libc-headers-k510 |grep -A3  "^do_install_armmultilib"
do_install_armmultilib() {
	oe_multilib_header asm/auxvec.h asm/bitsperlong.h asm/byteorder.h asm/fcntl.h asm/hwcap.h asm/ioctls.h  asm/kvm_para.h asm/mman.h asm/param.h asm/perf_regs.h asm/bpf_perf_event.h
	oe_multilib_header asm/posix_types.h asm/ptrace.h  asm/setup.h  asm/sigcontext.h asm/siginfo.h asm/signal.h asm/stat.h  asm/statfs.h asm/swab.h  asm/types.h asm/unistd.h
}
```

qemuarm64
```
$ bitbake -e linux-libc-headers-k510 |grep -A3  "^do_install_armmultilib" 
do_install_armmultilib() {
	oe_multilib_header asm/auxvec.h asm/bitsperlong.h asm/byteorder.h asm/fcntl.h asm/hwcap.h asm/ioctls.h asm/kvm.h asm/kvm_para.h asm/mman.h asm/param.h asm/perf_regs.h asm/bpf_perf_event.h
	oe_multilib_header asm/posix_types.h asm/ptrace.h  asm/setup.h  asm/sigcontext.h asm/siginfo.h asm/signal.h asm/stat.h  asm/statfs.h asm/swab.h  asm/types.h asm/unistd.h
}
```

Comment (hiraku-wfs) [reply recipes-kernel/linux-libc-headers/linux-libc-headers-k510.bb]: Got it. But we need more detailed comments like:

```
#
# For 5.10/arm, "asm/kvm.h" has been removed in the following kernel commit:
#     541ad0150ca4 [arm: Remove 32bit KVM host support]
#
# NOTE:
# It's the dirty hack that "_remove" a string for the function (it simply
# removes the matching string in the function), but it works and is simpler
# than rewriting do_install_armmultilib(). So we only adopt this method here.
# The oe-core upstream resolved this issue in other method:
#     4c3750bbc9da [libc-headers: update to v5.8]
#
do_install_armmultilib_remove_arm = "asm/kvm.h"
```

Comment (Ponsuke07) [reply recipes-kernel/linux-libc-headers/linux-libc-headers-k510.bb]: Apologies for the late reply. I have updated comment!

Comment (hiraku-wfs) [reply recipes-kernel/linux-libc-headers/linux-libc-headers-k510.bb]: No worries. Thank you!

Comment (hiraku-wfs) [reply recipes-kernel/linux-libc-headers/linux-libc-headers-k510.bb]: But GitHub says "This branch has conflicts that must be resolved", so could you rebase your commit onto the latest warrior branch?

Comment (Ponsuke07) [reply recipes-kernel/linux-libc-headers/linux-libc-headers-k510.bb]: OK, I've rebased.

Comment (hiraku-wfs) [reply recipes-kernel/linux-libc-headers/linux-libc-headers-k510.bb]: Thanks!

**Pull request merged**